Doc/reference/datamodel.rst

```diff
@@ -425,6 +425,14 @@ Mappings
       equal (e.g., ``1`` and ``1.0``) then they can be used interchangeably to index
       the same dictionary entry.
 
+      .. versionchanged:: 3.7
+         Dictionaries preserve insertion order, meaning that keys will be produced
+         in the same order they were added sequencially over the dictionary.
+         Replacing an existing key does not change the order, however removing a key
+         and re-inserting it will add it to the end instead of keeping its old place.
+         Note that older Python versions dictionaries do not preserve insertion order.
+         This behavior was an implementation detail of CPython from 3.6.
+
```

Comment:
Hmm, this isn't quite what I was going for.  My point was that *both* changes are needed: a description in the main portion stating how the insertion ordering works, and a `versionchanged` note about when that change was made.  Look at other `versionchanged` notes for examples.

Also, the `versionchanged` note should be at the end of the section, not in the middle.

Reply:
I looked at other versionchanged and tried to do as you've said, I hope it is correct.